### DISCUSSION TITLE
Multiple field arrays are duplicating values

### DISCUSSION THREAD

**Discussion by kodama007** [react-hook-form/react-hook-form#8715] [Q&A] [answered]
I currently have a component that I use to render field array. At the moment, I have two in my form, let's say apples and oranges.
The problem is that when I enter one apple in the array, the values all get duplicated to the oranges array.

```
const FormFieldsArray = ({ constructField, fields, control, page }) => {
    const handleChange = (panel) => (event, newExpanded) => {
        setExpanded(newExpanded ? panel : false);
    };

    const {
        fields: groupedFields,
        append,
        remove,
    } = useFieldArray({
        control,
        name: page.name,
    });

    return (
        <>
</>
);
```

Page name is getting changed from `apples` to `oranges`.

**Comment by Moshyfawn** [comment, marked as answer]
The field array `name` is "cached" on first render and is not dynamic by design. You're better off creating multiple different components with different `useFieldArray` hook calls and marking the actual input component reusable instead.

**Reply by e-compton** [reply]
So is it not possible at all to model a list within a list using react-hook-form?
